x-pack/plugins/fleet/server/services/agent_policy.ts

```diff
@@ -170,20 +170,36 @@ class AgentPolicyService {
         getAllowedOutputTypeForPolicy(existingAgentPolicy)
       );
     }
-    await soClient.update<AgentPolicySOAttributes>(SAVED_OBJECT_TYPE, id, {
-      ...agentPolicy,
-      ...(options.bumpRevision ? { revision: existingAgentPolicy.revision + 1 } : {}),
-      ...(options.removeProtection
-        ? { is_protected: false }
-        : { is_protected: agentPolicy.is_protected }),
-      updated_at: new Date().toISOString(),
-      updated_by: user ? user.username : 'system',
-    });
+    try {
+      await soClient.update<AgentPolicySOAttributes>(
+        SAVED_OBJECT_TYPE,
+        id,
+        {
+          ...agentPolicy,
+          ...(options.bumpRevision ? { revision: existingAgentPolicy.revision + 1 } : {}),
+          ...(options.removeProtection
+            ? { is_protected: false }
+            : { is_protected: agentPolicy.is_protected }),
+          updated_at: new Date().toISOString(),
+          updated_by: user ? user.username : 'system',
+        },
+        {
+          version: existingAgentPolicy.version,
```

Comment:
I tested this with an `oblt-cli` cluster and seeing the conflict error in the logs, which is good. I think this will prevent at least 2 instances overwriting the same revision with different content.

I'm going to find a way to simulate the concurrent 2 kibanas locally to be able to test, got some tips to use integration tests by kibana-core.

```
 [ingest-agent-policies:policy-elastic-agent-on-cloud]: version conflict, required seqNo [2071], primary term [1]. current document has seqNo [2073] and primary term [1]: version_conflict_engine_exception
	Root causes:
		version_conflict_engine_exception: [ingest-agent-policies:policy-elastic-agent-on-cloud]: version conflict, required seqNo [2071], primary term [1]. current document has seqNo [2073] and primary term [1]
```

Reply:
We have some similar integration tests for multiple Kibanas here: https://github.com/elastic/kibana/blob/main/x-pack/plugins/fleet/server/integration_tests/ha_setup.test.ts